### PR TITLE
Cleanup before 1.0 release

### PR DESCRIPTION
**Pull request by akshayjshah**
Addresses #166.

### CODE REVIEW

File: version.go [hunk 1]
```diff
@@ -23,4 +23,4 @@ package tchannel
 // VersionInfo identifies the version of the TChannel library.
 // Due to lack of proper package management, this version string will
 // be maintained manually.
-const VersionInfo = "0.0.1-dev"
+const VersionInfo = "1.0.0"
```
Comment (breerly) [inline version.go]: From the POV of someone using this library, how do they "install" `1.0.0`? Use the correlating commit hash?

Comment (akshayjshah) [inline version.go]: Typically, yes. Some of the build tools, e.g. Glide, also allow pinning to a tag.

Comment (breerly) [inline version.go]: Oh wow - we should def look into that.

**Comment by prashantv** [comment]
lgtm

**Comment by akshayjshah** [comment]
Okay, this should be ready to go now. Coveralls support is limited - the coverage report only covers the `github.com/uber/tchannel-go` package itself, not any of the sub-packages.

We should port over some of `go-build`'s Makefile fanciness to make linting, vetting, and coverage reporting better. I created #171 to track that task.

**Comment by prashantv** [comment]
Sweet, thanks for adding coverage. Yep, we can cover multiple packages coverage in a later change.

:rocket:

**Pull request merged**